lib/game.rb

```diff
@@ -0,0 +1,52 @@
+require_relative '../lib/game_state'
+require_relative '../lib/game_presenter'
+require_relative '../lib/messages'
+require_relative '../lib/board'
+require_relative '../lib/validator'
+require_relative '../lib/player'
+require_relative '../lib/game_settings'
+
+class Game
+    include Validator
+    include Messages
+
+    attr_accessor :game_state, :game_presenter
+
+    def initialize(game_presenter: ConsoleIO.new())
+        @game_presenter = game_presenter
+    end
+
+    def set_game_settings(settings:)
+        settings.each.reduce({}) do |map, setting_type|
+            input = nil
+            while !setting_type.is_valid?(input: input)
+                print_banner()
+                game_presenter.output_message(message: setting_type.message())
+                input = game_presenter.get_input() 
+            end
+            map[setting_type.name] = input
```

Comment:
maybe input should be "cleaned" by setting_type as well?
board size , should convert it to string
opponent type, can do nothing

so it doesnt need to be done in start(), when instantiating the new GameState class